File: webflux-dialogue/src/main/java/com/study/webflux/rag/application/controller/DashboardController.java
package com.study.webflux.rag.application.controller;

import java.net.URI;

import org.springframework.http.HttpStatus;
import org.springframework.http.server.reactive.ServerHttpResponse;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import reactor.core.publisher.Mono;

@RestController
public class DashboardController {

	@GetMapping("/dashboard")
	public Mono<Void> dashboard(ServerHttpResponse response) {
		response.setStatusCode(HttpStatus.TEMPORARY_REDIRECT);
		response.getHeaders().setLocation(URI.create("/dashboard.html"));
		return response.setComplete();
	}

	@GetMapping("/monitoring")
	public Mono<Void> monitoring(ServerHttpResponse response) {
		response.setStatusCode(HttpStatus.TEMPORARY_REDIRECT);
		response.getHeaders().setLocation(URI.create("/dashboard.html"));
		return response.setComplete();
	}
}

File: app-miyou/src/main/kotlin/com/miyou/app/infrastructure/inbound/web/monitoring/DashboardController.kt
package com.miyou.app.infrastructure.inbound.web.monitoring

import org.springframework.http.HttpStatus
import org.springframework.http.server.reactive.ServerHttpResponse
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController
import reactor.core.publisher.Mono
import java.net.URI

@RestController
class DashboardController {
    @GetMapping("/dashboard")
    fun dashboard(response: ServerHttpResponse): Mono<Void> {
        response.statusCode = HttpStatus.TEMPORARY_REDIRECT
        response.headers.location = URI.create("/admin/monitoring/grafana/")
        return response.setComplete()
    }

    @GetMapping("/monitoring")
    fun monitoring(response: ServerHttpResponse): Mono<Void> {
        response.statusCode = HttpStatus.TEMPORARY_REDIRECT
        response.headers.location = URI.create("/admin/monitoring/grafana/")
        return response.setComplete()
    }
}
